group/assignment5/src/AlphabetDiscoverer.java

```diff
@@ -17,9 +18,13 @@ private void buildGraphFromDictionary(List<String> dictionary) {
         if(dictionary.isEmpty()) {
             throw new IllegalArgumentException("Dictionary cannot be empty");
         }
-        graph = new Graph<>();
-        addAllCharsInDictionaryToGraph(dictionary);
-        addConstraints(dictionary);
+        if(!dictionary.equals(currentDictionary))
```

Comment:
Created this check so "buildGraphFromDictionary" won't be executed unnecessarily.